src/interceptors/XMLHttpRequest/XMLHttpRequestOverride.ts

```diff
@@ -322,6 +347,15 @@ export const createXMLHttpRequestOverride = (
               this.responseText = originalRequest.responseText
               this.responseXML = originalRequest.responseXML
 
+              const originalSymbols = Object.getOwnPropertySymbols(
+                originalRequest
+              )
+              const propertiesSymbol = originalSymbols[2]
+
+              const originalRequestProperties: XMLHttpRequestProperties =
```

Comment:
This is extremely hacky. I couldn't find a way to get the original XHR response headers. 

- `originalRequest.getAllResponseHeaders()` does not contain the actual headers that are at the same time available in the private symbol property on the same object.

Reply:
Nonetheless, a clever solution to access the original headers.

Reply:
I think @marcosvega91 has rightfully pointed out that I was not getting the response headers because there was a CORS policy in place and it doesn't propagate custom headers to the response by default.

That could be solved by adding the `allow-control-expose-headers` header on the response in a fake Express server. We shouldn't rely on internal properties behind symbols, I think that's highly subjective to bugs.